digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/process/ProjectDeltaCalculatorSpec.scala

```diff
@@ -272,15 +266,82 @@ class ProjectDeltaCalculatorSpec extends FunSuite with Matchers {
     }
   }
 
+  test("Test ProjectDeltaCalculator.partition " +
+       "When transferring a part of an existing roadpart to as another roadpart with more than one roadway" +
+       "Then continuity codes should be correct.") {
+    runWithRollback {
+      val originalDiscontinuity = Discontinuity.EndOfRoad
+      val newDiscontinuity      = Discontinuity.Discontinuous
+
+      /* A road of length 120 is split from 60 to two parts and discontinuity is changed. */
+      val addresses  = (0 to 9).map(i => {
+       createRoadAddress(i * 12, 12L)
+      })
+
+      val splitAddress = 60
+      val links        = addresses.filter(_.endAddrMValue <= splitAddress).map(a => {
+        toProjectLink(project, LinkStatus.UnChanged)(a).copy(roadwayId = 0)
+      })
+      val links2 = addresses.filter(a => {
+        a.endAddrMValue > splitAddress && a.endAddrMValue < 90
+      }).map(a => {
+        toProjectLink(project, LinkStatus.Transfer)(a).copy(roadwayId = 1, roadPartNumber = 206)
+      })
+                                                                          val links2a = addresses.filter(a => {
+        a.endAddrMValue > 90
+      }).map(a => {
+        toProjectLink(project, LinkStatus.Transfer)(a).copy(roadwayId = 2, roadPartNumber = 206)
+      })
+
+      /* The first part has one roadway, the seconds has two roadways. */
+      val roadway205_1 = toRoadway(links)
+      val roadway205_2 = toRoadway(links2).copy(id = 1)
+
+      val roadway205_3 = toRoadway(links2a).copy(id = 2, discontinuity = originalDiscontinuity)
+      roadwayDAO.create(Seq(roadway205_1, roadway205_2, roadway205_3))
+
+      val links3      = (links2 ++ links2a).map(pl => {
+        pl.copy(startAddrMValue = pl.startAddrMValue - splitAddress, endAddrMValue = pl.endAddrMValue - splitAddress)
+      })
+
+      val links4      = links3.init :+ links3.last.copy(discontinuity = newDiscontinuity)
+      val partitions  = ProjectDeltaCalculator.partitionWithProjectLinks(links ++ links4, links)
+      val partitions2 = partitions.adjustedSections.zip(partitions.originalSections)
+
+      partitions2.size should be(2)
+      val partitions3 = partitions2.partition(par => {
+        par._1.roadPartNumberStart == 205 && par._2.roadPartNumberStart == 205
+      })
+      partitions3._1 should have size 1
+      partitions3._2 should have size 1
+
+      /* Check continuities and road addresses. */
+      val (to1, fr1) = partitions3._1.head
```

Comment:
"to"? "fr"?

Tähän mennessä multa myös jo ehtii kadota käsitys siitä, mitä partitions2._1.head, ja alemman blokin partitions2._2.head sisältävät. -> BLokeille vielä kommentit siitä, _minkä_ jatkuvuudet ja osoitteet näissä tsekataan. 
(Tai sitten nimiremppaa, eli kuvaavammat nimet partitionsX- ym. muuttujille s.e. kertovat eksplisiittisesti mitä sisältävät.)